### PR TITLE
initial comparison and like calcs based on collation

### PR DESCRIPTION
**Pull request by bheni**
Currently the LIKE operator always does a case sensitive match.  This should be based on the collation.  This puts initial support in for doing per collation comparison and like calculations.  Currently the comparison changes are disabled as the codebase uses the compare function for more than just WHERE clause calculations which sometimes need to be case sensitive.

### CODE REVIEW

File: sql/expression/like_test.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ func TestLike(t *testing.T) {
 		{"a%b", "ab", true},
 		{"a%b", "a", false},
 		{"a_b", "ab", false},
+		{"aa:%", "AA:BB:CC:DD:EE:FF", true},
```
Comment (andrew-wm-arthur) [inline sql/expression/like_test.go:68]: this test could be simpler, why not
```
{"aa:%", "AA:BB", true},
```

File: sql/charsetcollation.go [hunk 2]
```diff
@@ -1200,10 +708,10 @@ type mysqlCollationRow struct {
...
 }
 
-var CollationToMySQLVals = map[Collation]mysqlCollationRow{
-	Collation_binary:             {63, Y, Y, 0, NoPad},
-	Collation_utf8_general_ci:    {33, Y, Y, 1, PadSpace},
-	Collation_utf8mb4_0900_ai_ci: {255, Y, Y, 0, NoPad},
+var CollationToMySQLVals = map[string]mysqlCollationRow{
+	Collation_binary.Name:             {63, Y, Y, 0, NoPad},
+	Collation_utf8_general_ci.Name:    {33, Y, Y, 1, PadSpace},
+	Collation_utf8mb4_0900_ai_ci.Name: {255, Y, Y, 0, NoPad},
```
Comment (andrew-wm-arthur) [inline sql/charsetcollation.go:714]: I think this is fine for now, given the size of the change already. I think we want `CollationToMySQLVals` and `characterSetMaxLengths`, etc. to be inlined into the Collation struct

**Review by andrew-wm-arthur — APPROVED** [review]
This looks good, Zach should take a pass as well

File: sql/expression/like_test.go (hunk 1 above)
Comment (bheni) [reply sql/expression/like_test.go:68]: no reason.  Was using a valid MAC address format like what we see in the nautobot code.

Comment (andrew-wm-arthur) [reply sql/expression/like_test.go:68]: ahh, got it

File: sql/charsetcollation.go (hunk 2 above)
Comment (bheni) [reply sql/charsetcollation.go:714]: Yeah, true.  Not exactly sure what I want to do with the collations structure / interface, but I'll go back and take a second look soon and figure out what to do about case sensitivity in where clauses which is busted atm.

**Pull request merged**